RawSpeed/LJpegDecompressor.cpp

```diff
@@ -463,27 +429,25 @@ void LJpegDecompressor::createHuffmanTable(HuffmanTable *htbl) {
  ************************************/
 
 void LJpegDecompressor::createBigTable(HuffmanTable *htbl) {
-  const uint32 bits = 14;      // HuffDecode functions must be changed, if this is modified.
+  const uint32 bits = HuffmanTable::TableBitDepth;      // HuffDecode functions must be changed, if this is modified.
   const uint32 size = 1 << bits;
-  int rv = 0;
-  int temp;
-  uint32 l;
 
   if (!htbl->bigTable)
     htbl->bigTable = (int*)_aligned_malloc(size * sizeof(int), 16);
```

Comment:
I believe, this is the closest to what i can provide: https://github.com/darktable-org/darktable/commit/38c69fb1b2bc90057c569242cb9945a10be0b583#diff-9eb11ed347ac11eae1731011513a742cR181
So i _think_ this is more or less what i was talking about: https://godbolt.org/g/JiOHFQ:
`test1()`  would overflow, `test2()` should be fine, while in `test3()` and `test4()` the order did not matter. Assuming i'm reading this right of course.